test/src/actions/PrintMaterials.C

```diff
@@ -0,0 +1,61 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "PrintMaterials.h"
+
+#include "ActionWarehouse.h"
+#include "AddMaterialAction.h"
+#include "MooseApp.h"
+
+#include <fstream>
+
+template<>
+InputParameters validParams<PrintMaterials>()
+{
+  InputParameters params = validParams<Action>();
+  params.addParam<FileName>("csv_file", "matinfo.csv", "CSV file for printing the material information");
+  return params;
+}
+
+PrintMaterials::PrintMaterials(const InputParameters & params) :
```

Comment:
Oh, I didn't see that this was in the tests directory. I can be alright with it in tests, I will look at it in more detail.